### PR TITLE
Add --keep-original-year flag to allow stale years to be unchanged

### PR DESCRIPTION
**Pull request by yiftachkarkason**
This allows use cases where the license needs to contain the creation time of the file (i.e. [like in Chromium](https://source.chromium.org/chromium/chromium/src/+/main:base/files/file_enumerator.cc)).

### CODE REVIEW

**Comment by Lucas-C** [comment]
Hi @yiftachkarkason!

Thank you for suggesting a new feature 😊

I wonder if this really neeeds a new optional parameter...
Shouldn't this be the default behaviour?
I'd be tempted to get rid of `match_years_strictly` in `insert_license.py` and just consider it always true.
What do you think?

Also, please run `black pre_commit_hooks/insert_license.py` in order for this file to be correctly formatted,
otherwise the GitHub Actions pipeline will fail.

**Comment by yiftachkarkason** [comment]
Hey @Lucas-C ! My PR actually uses `match_years_strictly=False` to allow past years to remain in the license, so I wouldn't remove `match_years_strictly` because it allows this flexibility

**Comment by Lucas-C** [comment]
> My PR actually uses `match_years_strictly=False` to allow past years to remain in the license, so I wouldn't remove `match_years_strictly` because it allows this flexibility

Alright. I made a few suggestions.
If you're fine with them, I'll be happy to merge this new feature!

**Comment by yiftachkarkason** [comment]
> > My PR actually uses `match_years_strictly=False` to allow past years to remain in the license, so I wouldn't remove `match_years_strictly` because it allows this flexibility
> 
> Alright. I made a few suggestions. If you're fine with them, I'll be happy to merge this new feature!

Great!
Where can I find the suggestions?

**Review by Lucas-C — CHANGES_REQUESTED** [review]

**Comment by Lucas-C** [comment]
> Where can I find the suggestions?

Sorry, I forgot to publish my review 😅
You should see them now

File: README.md [hunk 1]
```diff
@@ -109,6 +109,9 @@ headers:
   years they contain -- as if they used the year currently present in the
   license file.
 
+You can also use `--keep-original-year` to allow stale years to be unchanged. Using both `--keep-original-year`
```
Comment (yiftachkarkason) [inline README.md:112]:
```suggestion
You can also use `--allow-past-years` to allow stale years to be unchanged. Using both `--allow-past-years` 
```

**Comment by yiftachkarkason** [comment]
> > Where can I find the suggestions?
> 
> Sorry, I forgot to publish my review 😅 You should see them now

Cool suggestions! Applied them 😄

File: README.md [hunk 2]
```diff
@@ -109,6 +109,9 @@ headers:
   years they contain -- as if they used the year currently present in the
   license file.
 
+You can also use `--allow-past-years` to allow stale years to be unchanged. Using both `--allow-past-years` 
+and `--use-current-year` issues a year range as described above.
```
Comment (Lucas-C) [inline README.md:113]:
```suggestion
You can also use `--allow-past-years` to allow stale years to be unchanged.
Using both `--allow-past-years` and `--use-current-year` issues a year
range as described above.
```

**Comment by Lucas-C** [comment]
Merged! Thank you for your contribution @yiftachkarkason 👍

**Comment by Lucas-C** [comment]
This has been released in https://github.com/Lucas-C/pre-commit-hooks/releases/tag/v1.5.3

**Pull request merged**